src/mlpack/methods/lsh/lsh_search_impl.hpp

```diff
@@ -1040,110 +1040,39 @@ double LSHSearch<SortPolicy, MatType>::ComputeRecall(
-  ar & BOOST_SERIALIZATION_NVP(secondHashWeights);
-  ar & BOOST_SERIALIZATION_NVP(bucketSize);
-  // needs specific handling for new version
-
-  // Backward compatibility: in older versions of LSHSearch, the secondHashTable
-  // was stored as an arma::Mat<size_t>.  So we need to properly load that, then
-  // prune it down to size.
-  if (version == 0)
-  {
-    arma::Mat<size_t> tmpSecondHashTable;
-    ar & BOOST_SERIALIZATION_NVP(tmpSecondHashTable);
+  size_t tables;
+  if (cereal::is_saving<Archive>())
+    tables = secondHashTable.size();
+  ar(CEREAL_NVP(tables));
 
-    // The old secondHashTable was stored in row-major format, so we transpose
-    // it.
-    tmpSecondHashTable = tmpSecondHashTable.t();
-
-    secondHashTable.resize(tmpSecondHashTable.n_cols);
-    for (size_t i = 0; i < tmpSecondHashTable.n_cols; ++i)
-    {
-      // Find length of each column.  We know we are at the end of the list when
-      // the value referenceSet.n_cols is seen.
-
-      size_t len = 0;
-      for (; len < tmpSecondHashTable.n_rows; ++len)
-        if (tmpSecondHashTable(len, i) == referenceSet.n_cols)
-          break;
-
-      // Set the size of the new column correctly.
-      secondHashTable[i].set_size(len);
-      for (size_t j = 0; j < len; ++j)
-        secondHashTable[i](j) = tmpSecondHashTable(j, i);
-    }
-  }
-  else
-  {
-    size_t tables;
-    if (Archive::is_saving::value)
-      tables = secondHashTable.size();
-    ar & BOOST_SERIALIZATION_NVP(tables);
-
-    // Set size of second hash table if needed.
-    if (Archive::is_loading::value)
-    {
-      secondHashTable.clear();
-      secondHashTable.resize(tables);
-    }
-
-    ar & BOOST_SERIALIZATION_NVP(secondHashTable);
-  }
-
-  // Backward compatibility: old versions of LSHSearch held bucketContentSize
-  // for all possible buckets (of size secondHashSize), but now we hold a
-  // compressed representation.
-  if (version == 0)
-  {
-    // The vector was stored in the old uncompressed form.  So we need to shrink
-    // it.  But we can't do that until we have bucketRowInHashTable, so we also
-    // have to load that.
-    arma::Col<size_t> tmpBucketContentSize;
-    ar & BOOST_SERIALIZATION_NVP(tmpBucketContentSize);
-    ar & BOOST_SERIALIZATION_NVP(bucketRowInHashTable);
-
-    // Compress into a smaller vector by just dropping all of the zeros.
-    bucketContentSize.set_size(secondHashTable.size());
-    for (size_t i = 0; i < tmpBucketContentSize.n_elem; ++i)
-      if (tmpBucketContentSize[i] > 0)
-        bucketContentSize[bucketRowInHashTable[i]] = tmpBucketContentSize[i];
-  }
-  else
+  // Set size of second hash table if needed.
+  if (cereal::is_loading<Archive>())
   {
-    ar & BOOST_SERIALIZATION_NVP(bucketContentSize);
-    ar & BOOST_SERIALIZATION_NVP(bucketRowInHashTable);
+    secondHashTable.clear();
+    secondHashTable.resize(tables);
   }
 
-  ar & BOOST_SERIALIZATION_NVP(distanceEvaluations);
+  ar(CEREAL_NVP(secondHashTable));
```

Comment:
It is used later to resize the table but I totally agree with you, everything related to `tables` can be removed since cereal, handle the serialization of `std::map`